src/Symfony/EventListener/ErrorListener.php

```diff
@@ -34,4 +41,47 @@ protected function duplicateRequest(\Throwable $exception, Request $request): Re
 
         return $dup;
     }
+
+    public function onControllerArguments(ControllerArgumentsEvent $event): void
```

Comment:
is that code from Symfony? It looks quite heavy could you explain to me why you needed that?

Reply:
I didn't need to do this to get the original exception in my own implementation.

```--- /dev/null
+++ ../src/Action/ExceptionAction.php
@@ -20,8 +20,10 @@
 use ApiPlatform\Util\OperationRequestInitiatorTrait;
 use ApiPlatform\Util\RequestAttributesExtractor;
 use Symfony\Component\ErrorHandler\Exception\FlattenException;
+use Symfony\Component\HttpFoundation\Exception\RequestExceptionInterface;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;
 use Symfony\Component\Serializer\SerializerInterface;
 
 /**
@@ -46,12 +48,20 @@
     /**
      * Converts an exception to a JSON response.
      */
-    public function __invoke(FlattenException $exception, Request $request): Response
+    public function __invoke(\Throwable $exception, Request $request): Response
     {
         $operation = $this->initializeOperation($request);
-        $exceptionClass = $exception->getClass();
-        $statusCode = $exception->getStatusCode();
+        $exceptionClass = $exception::class;
+        $statusCode = 500;
+        $headers = [];
 
+        if ($exception instanceof HttpExceptionInterface) {
+            $statusCode = $exception->getStatusCode();
+            $headers = $exception->getHeaders();
+        } elseif ($exception instanceof RequestExceptionInterface) {
+            $statusCode = 400;
+        }
+
         $exceptionToStatus = array_merge(
             $this->exceptionToStatus,
             $operation ? $operation->getExceptionToStatus() ?? [] : $this->getOperationExceptionToStatus($request)
@@ -65,7 +75,6 @@
             }
         }
 
-        $headers = $exception->getHeaders();
         $format = ErrorFormatGuesser::guessErrorFormat($request, $this->errorFormats);
         $headers['Content-Type'] = sprintf('%s; charset=utf-8', $format['value'][0]);
         $headers['X-Content-Type-Options'] = 'nosniff';
```

This was all of the change that I needed to be able to get the actual exception.